### DISCUSSION TITLE
MCP23017 SPI

### DISCUSSION THREAD

**Discussion by jefersondossantos** [tttapa/Control-Surface#620] [Q&A] [answered]
Hello everybody.
The code for rotary encoders uses the mcp23017 i2c. Is it possible to use the spi version which is faster?

**Comment by tttapa** [comment, marked as answer]
This is not currently supported, but you can relatively easily add support for it:
You'll have to edit the following code:

https://github.com/tttapa/Control-Surface/blob/e6542665da8997fdf33aee5ff1e0b7383c7ebe00/src/AH/Hardware/MCP23017Encoders.hpp#L75-L91

https://github.com/tttapa/Control-Surface/blob/e6542665da8997fdf33aee5ff1e0b7383c7ebe00/src/AH/Hardware/MCP23017Encoders.hpp#L131-L169
